### PR TITLE
VS-282: Update Wrangler docs according to Vectorize V2 spec

### PR DESCRIPTION
**Pull request by garvit-gupta**
### Summary

VS-282: Update Wrangler docs according to Vectorize V2 spec. The current Wrangler docs are written according to the V1 spec and Vectorize V1 is being replaced by V2.

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/content/docs/workers/wrangler/commands.mdx [hunk 1]
```diff
@@ -613,15 +628,20 @@ wrangler vectorize get <INDEX_NAME>
...
 - `INDEX_NAME` string required
   - The name of the index to fetch details for.
+- `--deprecated-v1` boolean optional
+  - Get a legacy Vectorize index. Please note that legacy Vectorize indexes are on a [deprecation path](/vectorize/reference/transition-vectorize-legacy).
 
-### `list`
+### `info`
 
-List all Vectorize indexes in your account, including the configured dimensions and distance metric.
+Get some additional information about an individual index, including its current state.
```
Comment (netgusto) [inline src/content/docs/workers/wrangler/commands.mdx]: This is too vague to add value to that section. I reco we either remove it, or detail the actual additional information.

File: src/content/docs/workers/wrangler/commands.mdx [hunk 2]
```diff
@@ -593,15 +593,30 @@ Interact with a [Vectorize](/vectorize/) vector database.
...
 
 ```txt
-wrangler vectorize create <INDEX_NAME> [--dimensions=<NUM_DIMENSIONS>] [--metric=<DISTANCE_METRIC>]
+wrangler vectorize create <INDEX_NAME> [--dimensions=<NUM_DIMENSIONS>] [--metric=<DISTANCE_METRIC>] [--description=<DESCRIPTION>]
 ```
 
 - `INDEX_NAME` string required
-  - The name of the new index to create. Cannot be changed.
-- `--dimensions` number optional
+  - The name of the new index to create. Must be unique for an account and cannot be changed after creation.
```
Comment (netgusto) [inline src/content/docs/workers/wrangler/commands.mdx]: We should take the opportunity to document that we cannot reuse a deleted index name either.

**Review by sejoker — APPROVED** [review]

File: src/content/docs/workers/wrangler/commands.mdx [hunk 3]
```diff
@@ -593,15 +593,30 @@ Interact with a [Vectorize](/vectorize/) vector database.
...
+- `--description` string optional
+  - A description for your index.
+- `--deprecated-v1` boolean optional
+  - Create a legacy Vectorize index. Please note that legacy Vectorize indexes are on a [deprecation path](/vectorize/reference/transition-vectorize-legacy).
+
+### `list`
+
+List all Vectorize indexes in your account, including the configured dimensions and distance metric.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 4]
```diff
@@ -613,15 +628,20 @@ wrangler vectorize get <INDEX_NAME>
...
+- `--deprecated-v1` boolean optional
+  - Get a legacy Vectorize index. Please note that legacy Vectorize indexes are on a [deprecation path](/vectorize/reference/transition-vectorize-legacy).
 
-### `list`
+### `info`
 
-List all Vectorize indexes in your account, including the configured dimensions and distance metric.
+Get some additional information about an individual index, including the vector count and details about the last processed mutation.
 
 ```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 5]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
+```
+
+- `INDEX_NAME` string required
+  - The name of the Vectorize index for which metadata indexes needs to be fetched.
+
+### `delete-metadata-index`
+
+Disable metadata filtering on the specified property.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 6]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
+- `INDEX_NAME` string required
+  - The name of the Vectorize index from which vectors need to be deleted.
+- `--ids` array required
+  - List of ids corresponding to the vectors that must be deleted.
+
+### `create-metadata-index`
+
+Enable metadata filtering on the specified property.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 7]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
+- `INDEX_NAME` string required
+  - The name of the Vectorize index from which vectors need to be fetched.
+- `--ids` array required
+  - List of ids for which vectors must be fetched.
+
+### `delete-vectors`
+
+Delete vectors in a Vectorize index using the provided ids.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 8]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
+- `--namespace` string optional
+  - Query response to only include vectors from this namespace.
+- `--filter` string optional
+  - Filter vectors based on this metadata filter. Example: `'{ 'p1': 'abc', 'p2': { '$ne': true }, 'p3': 10, 'p4': false, 'nested.p5': 'abcd' }'`
+
+### `get-vectors`
+
+Fetch vectors from a Vectorize index using the provided ids.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 9]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
 - `--file` string required
   - A file containing the vectors to insert in newline-delimited JSON (JSON) format.
 - `--batch-size` number optional
   - The number of vectors to insert at a time (default: `5000`).
 
+### `query`
+
+Query a Vectorize index for similar vectors.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

File: src/content/docs/workers/wrangler/commands.mdx [hunk 10]
```diff
@@ -644,12 +666,117 @@ wrangler vectorize insert <INDEX_NAME> [OPTIONS]
...
+- `--batch-size` number optional
+  - The number of vectors to insert at a time (default: `1000`).
+- `--deprecated-v1` boolean optional
+  - Insert into a legacy Vectorize index. Please note that legacy Vectorize indexes are on a [deprecation path](/vectorize/reference/transition-vectorize-legacy).
+
+### `upsert`
+
+Upsert vectors into an index. Existing vectors in the index would be overwritten.
+
+```txt
```
Comment (kodster28) [inline src/content/docs/workers/wrangler/commands.mdx]:
```suggestion
```sh
```

**Review by kodster28 — APPROVED** [review]

**Review by KimJ15 — APPROVED** [review]

**Comment by SamRechHeng** [comment]
https://github.com/cloudflare/cloudflare-docs/pull/16469#pullrequestreview-2264023514

**Comment by SamRechHeng** [comment]
https://github.com/cloudflare/cloudflare-docs/pull/16469#pullrequestreview-2264027544

**Pull request merged**